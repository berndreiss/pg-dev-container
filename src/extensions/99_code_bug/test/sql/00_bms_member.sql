-- KO - Null parameter is not supported
SELECT check_bms_membership(NULL);

-- KO - Value out of INT range
SELECT check_bms_membership(99999999999999999);

-- KO - Nagative values are not supported
SELECT check_bms_membership(-100);

SELECT check_bms_membership(0);

SELECT check_bms_membership(10);

SELECT check_bms_membership(100);

