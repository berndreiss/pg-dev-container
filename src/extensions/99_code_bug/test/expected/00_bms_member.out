-- KO - Null parameter is not supported
SELECT count_via_bms(NULL);
 count_via_bms 
---------------
              
(1 row)

-- KO - Value out of INT range
SELECT count_via_bms(99999999999999999);
ERROR:  function count_via_bms(bigint) does not exist
LINE 1: SELECT count_via_bms(99999999999999999);
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
-- KO - Negative values are not supported
SELECT count_via_bms(-100);
ERROR:  max value must be greater than zero
SELECT count_via_bms(0);
 count_via_bms 
---------------
             0
(1 row)

SELECT count_via_bms(10);
 count_via_bms 
---------------
            10
(1 row)

SELECT count_via_bms(100);
 count_via_bms 
---------------
           100
(1 row)

-- Test debug statements
SET client_min_messages = DEBUG2;
SELECT count_via_bms(100);
DEBUG:  count_via_bms performed "101" iterations
 count_via_bms 
---------------
           100
(1 row)

RESET client_min_messages;
